src/platform/packages/shared/kbn-scout/src/playwright/runner/run_tests.ts

```diff
@@ -8,99 +8,140 @@
  */
 
 import { resolve } from 'path';
-import { exec } from 'child_process';
-import { promisify } from 'util';
-const execPromise = promisify(exec);
-
 import { ToolingLog } from '@kbn/tooling-log';
-import { withProcRunner } from '@kbn/dev-proc-runner';
+import { ProcRunner, withProcRunner } from '@kbn/dev-proc-runner';
 import { getTimeReporter } from '@kbn/ci-stats-reporter';
 import { REPO_ROOT } from '@kbn/repo-info';
 import { runElasticsearch, runKibanaServer } from '../../servers';
 import { loadServersConfig } from '../../config';
 import { silence } from '../../common';
 import { RunTestsOptions } from './flags';
 import { getExtraKbnOpts } from '../../servers/run_kibana_server';
-import { getPlaywrightGrepTag } from '../utils';
+import { getPlaywrightGrepTag, execPromise } from '../utils';
+import { ScoutPlaywrightProjects } from '../types';
+
+export const getPlaywrightProject = (
+  testTarget: RunTestsOptions['testTarget'],
+  mode: RunTestsOptions['mode']
+): ScoutPlaywrightProjects => {
+  if (testTarget === 'cloud') {
+    return mode === 'stateful' ? 'ech' : 'mki';
+  }
+
+  return 'local';
+};
+
+async function runPlaywrightTest(procs: ProcRunner, cmd: string, args: string[]) {
+  return procs.run(`playwright`, {
+    cmd,
+    args,
+    cwd: resolve(REPO_ROOT),
+    env: {
+      ...process.env,
+    },
+    wait: true,
+  });
+}
+
+export async function hasTestsInPlaywrightConfig(
+  log: ToolingLog,
+  cmd: string,
+  cmdArgs: string[],
+  configPath: string
+): Promise<boolean> {
+  log.info(`scout: Validate Playwright config has tests`);
+  try {
+    const validationCmd = `SCOUT_REPORTER_ENABLED=false ${cmd} ${cmdArgs.join(' ')} --list`;
+    log.debug(`scout: running '${validationCmd}'`);
+
+    const result = await execPromise(validationCmd);
+    const lastLine = result.stdout.trim().split('\n').pop() || '';
+
+    log.info(`scout: ${lastLine}`);
+    return true; // success
+  } catch (err) {
+    log.error(`scout: No tests found in [${configPath}]`);
+    return false; // failure
+  }
+}
```

Comment:
Moving to function that we can easily check after I found out it was silently broken

<img width="1421" alt="image" src="https://github.com/user-attachments/assets/8b32b2a8-cac8-463d-831a-13d6111cd48d" />